fix(build): nest start:dev fails with Cannot find module dist/main due to incremental+deleteOutDir interaction

## Bug

`npm run start:dev` (and `nest start --watch`) fails with:

```
Error: Cannot find module '/home/blur/erp2/backend/dist/main'
```

## Root Cause

Two settings interact badly:

- `nest-cli.json` has `"deleteOutDir": true` — watch mode deletes `dist/` on every restart
- `tsconfig.json` has `"incremental": true` — tsc uses `tsconfig.build.tsbuildinfo` to skip unchanged files

When `dist/` is deleted but `tsconfig.build.tsbuildinfo` still exists, tsc considers `main.js` up-to-date (per the cache) and skips emitting it. The result is a `dist/` with no `main.js`.

## Repro

```bash
cd backend
rm -rf dist
npm run start:dev   # fails: Cannot find module '.../dist/main'
```

Workaround: also delete the buildinfo before starting:
```bash
rm -f tsconfig.build.tsbuildinfo && npm run start:dev
```

## Fix

In `backend/tsconfig.build.json`, override `incremental` to `false`:

```json
{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "rootDir": "./src",
    "incremental": false
  },
  "include": ["src"],
  "exclude": ["src/modules/reports/**/*"]
}
```

This disables the incremental cache for the build config only, so `deleteOutDir` always gets a full emit. The base `tsconfig.json` can keep `incremental: true` for editor/IDE use.

--- a/backend/tsconfig.build.json
+++ b/backend/tsconfig.build.json
@@ -1,7 +1,8 @@
 {
   "extends": "./tsconfig.json",
   "compilerOptions": {
-    "rootDir": "./src"
+    "rootDir": "./src",
+    "incremental": false
   },
   "include": ["src"],
   "exclude": [

